src/ArduinoIoTCloudTCP.cpp

```diff
@@ -585,24 +726,33 @@ void ArduinoIoTCloudTCP::handleMessage(int length)
     bytes[i] = _mqttClient.read();
   }
 
+  /* Topic for OTA properties and device configuration */
+  if (_deviceTopicIn == topic) {
+    CBORDecoder::decode(_device_property_container, (uint8_t*)bytes, length);
+    DEBUG_VERBOSE("ArduinoIoTCloudTCP::%s [%d] device configuration or OTA message received", __FUNCTION__, millis());
+    _last_device_subscribe_cnt = 0;
+    _next_device_subscribe_attempt_tick = 0;
+  }
+
+  /* Topic for user input data */
   if (_dataTopicIn == topic) {
-    CBORDecoder::decode(_property_container, (uint8_t*)bytes, length);
+    CBORDecoder::decode(_thing_property_container, (uint8_t*)bytes, length);
   }
 
+  /* Topic for sync Thing last values on connect */
   if ((_shadowTopicIn == topic) && (_state == State::RequestLastValues))
   {
     DEBUG_VERBOSE("ArduinoIoTCloudTCP::%s [%d] last values received", __FUNCTION__, millis());
-    CBORDecoder::decode(_property_container, (uint8_t*)bytes, length, true);
-    sendPropertiesToCloud();
+    CBORDecoder::decode(_thing_property_container, (uint8_t*)bytes, length, true);
     _time_service.setTimeZoneData(_tz_offset, _tz_dst_until);
     execCloudEventCallback(ArduinoIoTCloudEvent::SYNC);
     _last_sync_request_cnt = 0;
     _last_sync_request_tick = 0;
-    _state = State::Connected;
+    _next_state = State::Connected;
```

Comment:
Is it really necessary to hold `_next_state` as a private member variable and therefore make it part of the overall state of an `ArduinoIoTCloudTCP` object? I'm sceptical of any additional member variable since each member variable equals at least two (bool) or many more possible states. Therefore many member variables equal exploding state space which makes it more easy to introduce new bugs and more hard to test all possible state transition paths.

Reply:
need to doublecheck, but i'm quite confident this commit can be dropped, because `State::CheckDeviceConfig` logic as changed after this commit.

Reply:
i've dropped this commit since at the end of the whole work it wasn't necessary.